sqlmesh/core/model/kind.py

```diff
@@ -152,8 +152,44 @@ def __repr__(self) -> str:
         return str(self)
 
 
+class OnDestructiveChange(str, Enum):
+    """What should happen when a forward-only model change requires a destructive schema change."""
+
+    ERROR = "ERROR"
+    WARN = "WARN"
+    ALLOW = "ALLOW"
+
+    @property
+    def is_error(self) -> bool:
+        return self == OnDestructiveChange.ERROR
+
+    @property
+    def is_warn(self) -> bool:
+        return self == OnDestructiveChange.WARN
+
+    @property
+    def is_allow(self) -> bool:
+        return self == OnDestructiveChange.ALLOW
+
+
+def _on_destructive_change_validator(
+    cls: t.Type, v: t.Union[OnDestructiveChange, str, exp.Identifier]
+) -> t.Any:
+    if v and not isinstance(v, OnDestructiveChange):
+        return OnDestructiveChange(v.this.upper() if isinstance(v, exp.Identifier) else v.upper())
+    return v
+
+
+on_destructive_change_validator = field_validator("on_destructive_change", mode="before")(
+    _on_destructive_change_validator
+)
+
+
 class _ModelKind(PydanticModel, ModelKindMixin):
     name: ModelKindName
+    on_destructive_change: OnDestructiveChange = OnDestructiveChange.ALLOW
```

Comment:
I think I've removed it in my PR. Why do we need this here if the property only applies to incremental models?